libcontainer/cgroups/utils.go

```diff
@@ -399,16 +403,19 @@ func WriteCgroupProc(dir string, pid int) error {
 	return err
 }
 
-// Since the OCI spec is designed for cgroup v1, in some cases
-// there is need to convert from the cgroup v1 configuration to cgroup v2
-// the formula for cpuShares is y = (1 + ((x - 2) * 9999) / 262142)
+// ConvertCPUSharesToCgroupV2Value converts a CPU shares cgroup v1 configuration
+// to cgroup v2.
+//
+// The OCI spec is designed for cgroup v1, in some cases there is need to
+// convert from the cgroup v1 configuration to cgroup v2 the formula for cpuShares
```

Comment:
The "in some cases" was a generic comment (as in "in some cases we need to convert from cgroup v1 to cgroup v2, and cpu shares is one such case").

The comment was talking about generic stuff first, when went to specifics of CPU shares.

I am not sure how to rephrase this in a good way if we must start with specifics (`ConvertCPUSharesToCgroupV2Value`). Maybe just skip the generic part?

```suggestion
// ConvertCPUSharesToCgroupV2Value converts a CPU shares value
// from cgroup v1 to cgroup v2.
//
// The OCI spec is designed for cgroup v1, in some cases there is need to
// convert from the cgroup v1 configuration to cgroup v2. The formula for cpuShares
// is y = (1 + ((x - 2) * 9999) / 262142), which is a conversion from
// [2-262144] to [1-10000].
```

or some such.

(note you did not separated the two sentences, the second one starts with "the formula for")

Reply:
OK I think the generic part (in some cases we need to convert) should be its own comment,
e.g.

```go
// The OCI spec is designed for cgroup v1, in some cases there is need to
// convert from the cgroup v1 configuration to cgroup v2. A few functions
// below are doing such conversion.

// ConvertCPUSharesToCgroupV2Value converts a CPU shares value
// from cgroup v1 to cgroup v2.
//
// The formula for CPU shares is ....
func ConvertCPUSharesToCgroupV2Value(cpuShares uint64) uint64 {
...

//  ConvertMemorySwapToCgroupV2Value converts MemorySwap value from OCI spec
// ...
func ConvertMemorySwapToCgroupV2Value(memorySwap, memory int64) (int64, error) {
...

// ConvertBlkIOToIOWeightValue converts a BlkIOWeight cgroup v1 value
// to cgroup v2.
//
// The formula for BlkIOWeight to IOWeight is y = (1 + (x - 10) * 9999 / 990),
// as we convert linearly from [10-1000] to [1-10000].
func ....
```
and so on